### PR TITLE
Remove boost.lambda dependency

### PR DESCRIPTION
**Pull request by mborland**

### CODE REVIEW

File: include/boost/math/tools/polynomial.hpp [hunk 1]
```diff
@@ -338,13 +327,12 @@ class polynomial
       }
    }
 #ifdef BOOST_MATH_HAS_IS_CONST_ITERABLE
```
Comment (jzmaddock) [inline include/boost/math/tools/polynomial.hpp:329]: I suspect BOOST_MATH_HAS_IS_CONST_ITERABLE is always set in C++11 land?

Other than that, this all looks good to me,, thanks!

Comment (mborland) [reply include/boost/math/tools/polynomial.hpp:329]: BOOST_MATH_HAS_IS_CONST_ITERABLE requires cxx14_variable_templates so it has to stay put for now.

Comment (jzmaddock) [reply include/boost/math/tools/polynomial.hpp:329]: Ah, shame.  No matter.

**Comment by mborland** [comment]
This one should also be good to go. CI is clean.

**Pull request merged**